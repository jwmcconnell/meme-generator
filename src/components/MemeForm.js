import React from 'react';
import PropTypes from 'prop-types';

import InputField from './InputField';

const MemeForm = ({ 
  updateFile,
  handleUpdate
}) => {

  const fieldsInfo = [
    { label: 'File Upload', type: 'file', update: updateFile },
    { label: 'URL', name: 'imageUrl', type: 'text', update: handleUpdate },
    { label: 'Top Text', name: 'topText', type: 'text', update: handleUpdate },
    { label: 'Bottom Text', name: 'bottomText', type: 'text', update: handleUpdate },
    { label: 'Font Size', name: 'fontSize', type: 'number', update: handleUpdate, min: '10', defaultValue: '26' },
    { label: 'Font Color', name: 'fontColor', type: 'color', update: handleUpdate, defaultValue: '#FFFFFF' },
  ];

  const fields = fieldsInfo.map(({ label, name, type, update, min, defaultValue }) => (
    <InputField 
      label={label} 
      type={type} 
      update={update} 
      name={name}
      min={min}
      defaultValue={defaultValue}
      key={`${label}-${type}-${update}`}
    />
  ));
  
  return (
    <section>
      {fields}

      <fieldset>
        <legend>Image Source</legend>

        <label htmlFor="upload">Upload
          <input 
            defaultChecked 
            required 
            name="imageOption" 
            value="upload" 
            type="radio" 
            id="upload"
            onChange={handleUpdate}
          />
        </label>

        <label htmlFor="url">URL
          <input 
            required 
            name="imageOption" 
            value="url" 
            type="radio" 
            id="url"
            onChange={handleUpdate}
          />
        </label>
            
      </fieldset>
    </section>
  );
};

MemeForm.propTypes = {
  updateFile: PropTypes.func.isRequired,
  handleUpdate: PropTypes.func.isRequired
};


export default MemeForm;
