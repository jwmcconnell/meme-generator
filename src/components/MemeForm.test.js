/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import MemeForm from './MemeForm';

describe('MemeForm component', () => {
  it('renders MemeForm', () => {
    
    const updateFile = e => console.log(e.target.value);
    const updateUrl = e => console.log(e.target.value);
    const updateImageOption = e => console.log(e.target.value);
  
    const updateTopText = e => console.log(e.target.value);
    const updateBottomText = e => console.log(e.target.value);
  
    const updateFontSize = e => console.log(e.target.value);
    const updateFontColor = e => console.log(e.target.value);
    
    const wrapper = shallow(
      <MemeForm 
        updateFile={updateFile}
        updateUrl={updateUrl}
        updateImageOption={updateImageOption}
        updateTopText={updateTopText}
        updateBottomText={updateBottomText}
        updateFontSize={updateFontSize}
        updateFontColor={updateFontColor}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
  
