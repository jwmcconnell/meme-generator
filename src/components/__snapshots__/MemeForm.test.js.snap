// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemeForm component renders MemeForm 1`] = `
<section>
  <InputField
    key="File Upload-file-function updateFile(e) {
      return console.log(e.target.value);
    }"
    label="File Upload"
    name="fileUpload"
    type="file"
    update={[Function]}
  />
  <InputField
    key="URL-text-function handleUpdate(e) {
      return console.log(e.target.value);
    }"
    label="URL"
    name="imageUrl"
    type="text"
    update={[Function]}
  />
  <InputField
    key="Top Text-text-function handleUpdate(e) {
      return console.log(e.target.value);
    }"
    label="Top Text"
    name="topText"
    type="text"
    update={[Function]}
  />
  <InputField
    key="Bottom Text-text-function handleUpdate(e) {
      return console.log(e.target.value);
    }"
    label="Bottom Text"
    name="bottomText"
    type="text"
    update={[Function]}
  />
  <InputField
    defaultValue="26"
    key="Font Size-number-function handleUpdate(e) {
      return console.log(e.target.value);
    }"
    label="Font Size"
    min="10"
    name="fontSize"
    type="number"
    update={[Function]}
  />
  <InputField
    defaultValue="#FFFFFF"
    key="Font Color-color-function handleUpdate(e) {
      return console.log(e.target.value);
    }"
    label="Font Color"
    name="fontColor"
    type="color"
    update={[Function]}
  />
  <label
    htmlFor="fontFamily"
    style={
      Object {
        "marginTop": ".8rem",
      }
    }
  >
    Font:
    <select
      id="fontFamily"
      name="fontFamily"
      onChange={[Function]}
    >
      <option
        defaultValue={true}
        value="Roboto"
      >
        Roboto
      </option>
      <option
        value="Arial"
      >
        Arial
      </option>
      <option
        value="Courier"
      >
        Courier
      </option>
      <option
        value="Times"
      >
        Times
      </option>
      <option
        value="Helvetica"
      >
        Helvetica
      </option>
    </select>
  </label>
  <fieldset>
    <legend>
      Image Source
    </legend>
    <label
      htmlFor="upload"
    >
      Upload
      <input
        defaultChecked={true}
        id="upload"
        name="imageOption"
        onChange={[Function]}
        required={true}
        type="radio"
        value="upload"
      />
    </label>
    <label
      htmlFor="url"
    >
      URL
      <input
        id="url"
        name="imageOption"
        onChange={[Function]}
        required={true}
        type="radio"
        value="url"
      />
    </label>
  </fieldset>
</section>
`;
